window/src/os/macos/app.rs

```diff
@@ -120,6 +129,26 @@ extern "C" fn application_open_file(
     }
 }
 
+extern "C" fn application_dock_menu(
+    self_: &mut Object,
+    _sel: Sel,
+    _app: *mut Object,
+) -> *mut Object {
+    unsafe {
+        let dock_menu = NSMenu::alloc(nil)
```

Comment:
Please implement this using the higher level menu wrappers found in:
https://github.com/wez/wezterm/blob/main/window/src/os/macos/menu.rs

You should be able to do something like:

```rust
let dock_menu = Menu::new_with_title("");

let new_window = MenuItem::new_with(
    "New Window",
    Some(sel!(weztermPerformKeyAssignment:)),
    "");
new_window.set_represented_item(
    RepresentedItem::KeyAssignment(KeyAssignment::SpawnWindow));
dock_menu.add_item(new_window);
dock_menu.autorelease()
```

and then you can delete the `wezterm_spawn_window` function.

Note: I think you may need to tweak `Menu::autorelease` to have the same signature as the underlying https://docs.rs/objc/latest/objc/rc/struct.StrongPtr.html#method.autorelease method that it calls for this to work

Reply:
I amended the commit so that it uses the existing wrapper API instead of direct use of `cocoa::appkit::{NSMenu, NSMenuItem}` which requires to be `unsafe`.